lib/src/lis/protocol.cpp

```diff
@@ -35,6 +35,28 @@ constexpr const int lis::tape_trailer::size;
 constexpr const int lis::reel_header::size;
 constexpr const int lis::reel_trailer::size;
 
+int sizeof_type(int reprc) noexcept (false) {
+    using repcode = lis::representation_code;
+    const auto code = static_cast< repcode >(reprc);
+
+    switch( code ) {
```

Comment:
I am not fond of repeat of the part of the logic in dfs_fmtstr, but using this method there might reduce readability, so I leave the decision to you.